Src/PoissonRecon.h

```diff
@@ -510,8 +511,8 @@ template<typename Real>
 void PoissonRecon<Real>::evaluate()
 {
     m_profiler.start();
-    double valueSum = 0;
```

Comment:
Yes, it was due to conversion warning at `m_isoValue = valueSum / weightSum;`

It sums at most number-of-points elements, so probably 100M is a reasonable enough upper bound. The value/weight are (I think) in the order of the point cloud resolution. The final iso-value is also in the order of the resolution, so both sums should **not** be more than a few orders of difference, so the overall sum, doesn't require high precision. If the resolution is very low, then I think the precision is dominated by the type of the values rather than the type of the accumulator.

So I don't think there should be a problem, but I also see no reason not to go on the safe side and perform the accumulation with doubles and only cast the result of the division.

Only drawback I do see to this is that we are making an assumption regarding `Real` (a weak one, but still). Do you think it is reasonable for someone to try to run the reconstruction with some "non-standard" number class? This was my original motivation for changing the types, rather than doing a cast.